apps/next/src/components/ui/nutrition-breakdown.tsx

```diff
@@ -0,0 +1,121 @@
+}
+
+interface Props {
+  dateString: string,
+  mealsEaten: LoggedMealJoinedWithNutrition[]
+}
+
+const NutritionBreakdown = ({ dateString, mealsEaten }: Props) => {
+  const nutrition: NutritionData = compileMealData(mealsEaten);
+
+  const utils = trpc.useUtils();
+  const deleteLoggedMealMutation = trpc.nutrition.deleteLoggedMeal.useMutation({
+      onSuccess: () => {
+        alert(`Removed dish from your log`);
+        utils.nutrition.invalidate();
+      },
+      onError: (error: Error) => {
+        alert(error.message);
+      }
+    });
+
+  const removeBtnOnClick = (e: React.MouseEvent, userId: string | null, dishId: string | null) => {
+    e.preventDefault();
+    if (!userId || !dishId) return;
+
+    if (!DUMMY_USER_ID) {
+      alert("You must be logged in to track meals");
+      return;
+    }
+
+    deleteLoggedMealMutation.mutate({ userId, dishId });
+  };
+
+  return (
+    <div>
+      <center style={{ fontSize: '2rem', fontWeight: 'bold' }}>{dateString}</center>
+      <div className="flex align-items mt-4">
+        <div className="flex flex-col">
+          <center style={{ fontSize: '1.25rem', fontWeight: 'bold' }}>Calories</center>
+          <ProgressDonut progress_value={nutrition.calories} max_value={2000} display_unit="" />
+        </div>
+        <div className="flex flex-col">
+          <center style={{ fontSize: '1.25rem', fontWeight: 'bold' }}>Protein</center>
+          <ProgressDonut progress_value={nutrition.protein_g} max_value={75} display_unit="g" />
+        </div>
+        <div className="flex flex-col">
+          <center style={{ fontSize: '1.25rem', fontWeight: 'bold' }}>Carbs</center>
+          <ProgressDonut progress_value={nutrition.carbs_g} max_value={250} display_unit="g" />
+        </div>
+        <div className="flex flex-col">
+          <center style={{ fontSize: '1.25rem', fontWeight: 'bold' }}>Fat</center>
+          <ProgressDonut progress_value={nutrition.fat_g} max_value={50} display_unit="g" />
+        </div>
+      </div>
+      <div className="meal-history">
+        {mealsEaten && mealsEaten.map((meal) => (
+          <div key={meal.userId + meal.dishId} className="flex items-center justify-between gap-4 rounded-lg border p-4 mb-3">
+            <div className="flex flex-col">
+              <h3 className="font-medium">{meal.servings} serving{meal.servings > 1 ? 's' : ''} of {meal.dishName}</h3>
+              <p className="text-sm text-muted-foreground">
+                {Math.round(meal.calories * meal.servings)} calories |&nbsp;
+                {Math.round(meal.protein * meal.servings)}g protein |&nbsp;
+                {Math.round(meal.carbs * meal.servings)}g carbs |&nbsp;
+                {Math.round(meal.fat * meal.servings)}g fat
+              </p>
+            </div>
+
+            <button className="h-8 rounded-md border px-3 text-sm transition-colors hover:bg-accent hover:text-accent-foreground" onClick={(e) => removeBtnOnClick(e, meal.userId, meal.dishId)}>
```

Comment:
Way too long for single line. Split the props up when it wraps over 80 characters.

```suggestion
            <button 
              className="h-8 rounded-md border px-3 text-sm transition-colors hover:bg-accent hover:text-accent-foreground" 
              onClick={(e) => removeBtnOnClick(e, meal.userId, meal.dishId)}
            >
```